### ISSUE TITLE
Android: Touch Exploration Mode state & Detect state change

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30865]
## Description
Android allows you to detect whether or not any accessibility service is running that uses touch exploration mode. This used to be more important, due to older versions of Talkback having this as an optional user setting, but since it is still supported in the API, theres no harm in adding a method to check it if desired.

This will generally be a better thing to check than `isScreenReaderEnabled` if you are only worried about whether or not you need to handle focus-navigation related differences, and not something specific to screen readers, such as audio feedback.

## React Native version:
v0.63

## Expected Behavior
The AccessibilityInfo API should have a method to detect if touch exploration mode is enabled, and a callback that can be triggered when its state changes.

## Android Details
The AccessibilityManager class has methods to detect both whether touch exploration is enabled as well as a method to run a callback when its state changes.

https://developer.android.com/reference/android/view/accessibility/AccessibilityManager#isTouchExplorationEnabled()

https://developer.android.com/reference/android/view/accessibility/AccessibilityManager#addTouchExplorationStateChangeListener(android.view.accessibility.AccessibilityManager.TouchExplorationStateChangeListener,%20android.os.Handler)

**Comment by blavalla** [comment]
Closed since this is no longer necessary now that we have the ability to detect enabled services.
